test/gtest/ucp/test_ucp_reconfigure.cc

```diff
@@ -0,0 +1,272 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2001-2015. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "ucp_test.h"
+#include "common/test.h"
+
+extern "C" {
+#include <ucp/wireup/address.h>
+#include <ucp/wireup/wireup_ep.h>
+#include <ucp/core/ucp_ep.inl>
+}
+
+class test_ucp_reconfigure : public ucp_test {
+protected:
+    using address_pair_t = std::pair<void*, ucp_unpacked_address_t>;
+    using address_pair_p =
+            std::unique_ptr<address_pair_t, void (*)(address_pair_t*)>;
+
+    class entity : public ucp_test_base::entity {
+    public:
+        entity(const ucp_test_param &test_params, ucp_config_t *ucp_config,
+               const ucp_worker_params_t &worker_params,
+               const ucp_test *test_owner) :
+            ucp_test_base::entity(test_params, ucp_config, worker_params,
+                                  test_owner),
+            m_cfg_index(UCP_WORKER_CFG_INDEX_NULL)
+        {
+        }
+
+        void connect(const ucp_test_base::entity *other,
+                     const ucp_ep_params_t &ep_params, int ep_idx = 0,
+                     int do_set_ep = 1) override;
+
+        void verify(const entity &other) const;
+
+        bool is_reconfigured() const
+        {
+            return m_cfg_index != ep()->cfg_index;
+        }
+
+        static const entity &to_reconfigured(const ucp_test_base::entity &e)
+        {
+            return *static_cast<const entity*>(&e);
+        }
+
+    private:
+        void store_config();
+        ucp_tl_bitmap_t ep_tl_bitmap() const;
+        address_pair_p get_address(bool ep_only) const;
+        bool has_matching_lane(ucp_ep_h ep, ucp_lane_index_t lane_idx,
+                               const entity &other) const;
+
+        ucp_worker_cfg_index_t m_cfg_index;
+        std::vector<uct_ep_h>  m_uct_eps;
+    };
+
+    void init() override
+    {
+        ucp_test::init();
+
+        /* Check presence of IB devices using rc_verbs, as all devices must
+         * support it. */
+        if (!has_resource(sender(), "rc_verbs")) {
+            UCS_TEST_SKIP_R("IB transport is not present");
+        }
+
+        m_entities.clear();
+    }
+
+public:
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_values(variants, get_test_variants_feature, 0);
+        add_variant_values(variants, get_test_variants_feature, 1, "excl_if");

```

Comment:
Clearer name (`exclude_iface`?)

Reply:
I think better keep variant names short, to avoid too long lines when gtest prints the full test name (possibly combined with other variant axis)